### PR TITLE
WIP: Add Kubernetes Tutorial

### PR DESCRIPTION
**Pull request by ahawkins**
Abstract: Tutorial inspired by deploying to Heroku. Readers will learn to connect their Buildkite agents to different types of Kubernetes clusters. No assumptions are made about the application, other than it’s a Docker image. Deployment happens via simple `kubectl` apply. That should demonstrate that Buildkite and Kubernetes work well enough together and gives readers the confidence to implement their own deployment process.

**TODO**

- [x] Link to tutorial in sidebar
- [x] Cover AWS & EKS
- [x] Covers AWS & self provisioned clusters (ala something like Kops
- [x] GCP & GKE
- [ ] "Any other cluster" grab bag.

### CODE REVIEW

File: pages/tutorials/connecting_kubernetes.md.erb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+### Configuring the IAM Role
+
+You'll need to grant EKS access to the IAM role associated with
+your agents. Attaching an existing AWS Managed Policy is the easiest
+way. The `PowerUser` policy is easiest for this tutorial since it
+grant access to all AWS APIs execpt IAM management.
+
+If you're using the ElasticStack, then open the CloudFormation console and set
+the `ManagedPolicyARN` parameter to `arn:aws:iam::aws:policy/PowerUserAccess`.
```
Comment (ahawkins) [inline pages/tutorials/connecting_kubernetes.md.erb]: Howe can I prevent emojification of these ARNs? They're being replaced with `` tags inside a code tag.

File: pages/tutorials/connecting_kubernetes.md.erb [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+```
+
+This command opens the ConfigMap in your editor. You'll see something
+like this:
+
+```yml
+apiVersion: v1
+data:
+  mapRoles: |
+    - rolearn: arn:aws:iam::555555555555:role/buidlkite-tutorial-worker-nodes-NodeInstanceRole-74RF4UBDUKL6
```
Comment (ahawkins) [inline pages/tutorials/connecting_kubernetes.md.erb]: Same emojification issue here.

File: pages/tutorials/connecting_kubernetes.md.erb [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      username: system:node:{{EC2PrivateDNSName}}
+      groups:
+        - system:bootstrappers
+        - system:nodes
+  mapUsers: |
+    - userarn: arn:aws:iam::555555555555:user/admin
```
Comment (ahawkins) [inline pages/tutorials/connecting_kubernetes.md.erb]: Same emojification issue here

File: pages/tutorials/connecting_kubernetes.md.erb [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        - system:masters
+```
+
+Add a new entry to the `mapRoles` list for the ARN from the previous
+step. Use the `system:masters` to grant access to all Kubernetes APIs.
+The snippet should look like:
+
+```yml
+# set rolearn to the ARN
+- rolearn: arn:aws:iam::828070532471:role/buildkite-kubernetes-tutorial-elastic-stack-Role
```
Comment (ahawkins) [inline pages/tutorials/connecting_kubernetes.md.erb]: Emojification issue here as well

File: pages/tutorials/migrating_from_bamboo.md.erb [hunk 5]
```diff
@@ -64,7 +64,7 @@ steps:
     artifact_paths: "build/*"
 
   # To prevent the "make test" stage from running before "make" has finished,
-  # we seperate the command with a "wait" step.
+  # we separate the command with a "wait" step.
```
Comment (ahawkins) [inline pages/tutorials/migrating_from_bamboo.md.erb]: My spelling mistake pre-commit hook picked this up. Let me know if you'd like me to cherry pick into another PR.

**Comment by harrietgrace** [comment]
Welcome @ahawkins! 👋🏻 We'll sort out access to those builds for you. It's failing at the moment because of an incomplete link to another docs page - if you update /agents/v3/hooks to /docs/agents/v3/hooks, it should pass 😊

File: pages/tutorials/connecting_kubernetes.md.erb (hunk 1 above)
Comment (harrietgrace) [reply pages/tutorials/connecting_kubernetes.md.erb]: If you escape the colons, it should sort itself out! I just tested out this: `arn\:aws\:iam:\:aws\:policy/PowerUserAccess` and it came up emoji-free.

File: pages/tutorials/migrating_from_bamboo.md.erb (hunk 5 above)
Comment (harrietgrace) [reply pages/tutorials/migrating_from_bamboo.md.erb]: Rad, thanks! It'll be fine in with the rest 😊

**Comment by toolmantim** [comment]
I've just hooked up a Heroku Review App setup to the docs repo here, so we can preview this live: https://buildkite-docs-review-pr-458.herokuapp.com/docs/tutorials/connecting-kubernetes

**Comment by ahawkins** [comment]
Closing in favor of #469

**Pull request closed without merging**